### ISSUE TITLE
Smoothness issue in fullscreen: parts of screen not updated ?

### ISSUE THREAD

**Issue by micler** [dosbox-staging/dosbox-staging#3820]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

0.81.1

### What Operating System are you using?

Windows 10

### Whats your question and how can we help?

Hi,
I'm having smoothness issues when running F1GP (microprose), and it happens in fullscreen only. I know it's not a cycles issues because it seems frames are not rendered: in this game you choose a refresh rate (here 30fps), if the CPU is overloaded, the whole game slows down. (it has a Processor occupancy build in it, it's around 30% with 20000 cycles).
I've tried several options to see if it had an impact (chaning the renderer, activating Vsync, locking the frame rate, etc...), to no avail.

I've recorded the same short scene twice: first in windowed mode, then in fullscreen, and attached the video.
https://github.com/user-attachments/assets/4aa8350c-61ef-4672-9fb3-9afb303b2eb2

On this scene it's not striking when playing the video (i had other clips when the issue is plainly visible, but no comparison to do), but looking at the images frames by frames, the issue can be seen: images not or only partly updated. It happens very rarely in windowed mode (i guess it's a synchronizing issue with the recorder), but very often in fullscreen.
I tried recording at 60fps (twice the frame rate of the game) to see it it helps, the issue is even more visible when looking at the frames: i can get up to 5-6 almost identical frames in a row.

It seems like the display buffer is not updated correctly.
Hardware is Ryzen 3600 + RX5700

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by MasterO2** [comment]
@micler, https://github.com/dosbox-staging/dosbox-staging/wiki/Game-issues#game-is-choppy-in-full-screen-mode perhaps?

**Comment by micler** [comment]
Already tried it, the video recorded has been done with glshader = none.

**Comment by Grounded0** [comment]
Unlikely to be a performance issue with a system like that.

**Comment by interloper98** [comment]
Can you post your console log output when playing 100% in window-mode and then a second log when playing 100% fullscreen? 

> it happens in fullscreen only

That's very critical information. I've looked at the presentation code, and there's nothing in it that will withhold frames or do partial rendering when fullscreen versus in window-mode.

However, there is some dynamic throttling that happens if the host bogs down and frames aren't rendered quickly enough - however, that behavior is always present regardless if you're windowed or fullscreen.

So my guess is there is some extra strain happening when you're fullscreen.

I would suggest testing with this config. It's specifically meant to take hardware scaling pressure off when fullscreen (might have to adapt the `[autoexec]` section for your installation):

```ini
[sdl]
fullscreen = on
fullresolution = 1280x720
presentation_mode = auto
output = texture
vsync = off

[dosbox]
dos_rate = 60

[cpu]
core = dynamic
cpu_cycles = 50000
cpu_throttle = yes

[midi]
mididevice = mt32

[sblaster]
sbtype = sbpro2
oplmode = none

[speaker]
pcspeaker = off

[joystick]
joysticktype = disabled

[ipx]
ipx = off

[ethernet]
ne2000 = off

[voodoo]
voodoo = off

[autoexec]
mount c .
mount a floppy -t floppy
imgmount d CD/F1GP.cue -t cdrom
c:
call run.bat
```

**Comment by micler** [comment]
> Unlikely to be a performance issue with a system like that.
> 
> Do you have variable refresh rate (VRR) enabled?

I tried both auto mode (I get 70-71fps, seen by my monitor, so I guess it's VRR) or fixed at 60fps, same issue.

**Comment by micler** [comment]
I'll try you configuration and send feedback.
I downloaded FPS  monitor to try see if the FPS rendered in dosbox were dropping, and to my surprise, having the overlay displayed made the issue disappear. That's so confusing. Feels like the overlay forced redraws.

**Comment by micler** [comment]
My config file at the moment, for comparison:
fullscreen          = false
fullresolution      = 640x400
windowresolution    = 640x400
vsync               = on
presentation_mode   = auto
output              = texture
texture_renderer    = auto

dos_rate                    = default

core      = auto
cputype   = auto
cycles    = 20000
cycleup   = 2000
cycledown = 2000

**Comment by interloper98** [comment]
@micler, seeing your fullscreen=640x400 setting has me suspecting you're using a CRT or perhaps old 4:3 display?

Looking at the code, maybe it demands all duplicate frames. So you could try forcing constant frame rate with:

```ini
[sdl]
presentation_mode = cfr
```

**Comment by johnnovak** [comment]
@micler To troubleshoot issues like this, it is important to know what monitor, GPU and driver settings you use.

So let's start by answering the following, please:

- Are you on a desktop PC or a laptop?
- Do you use a single external monitor? (If on a desktop PC)
- Make and model number of your monitor (or at the very least state if it's VRR or not)
- Model number of your GPU
- Are you forcing any non-standard GPU setting in the driver globally or just for Staging? (E.g. forcing vsync, forcing triple buffering)
 
Also, please try a few more games in Staging and let us know if this a systemic issue or just limited to this single game.

**Comment by johnnovak** [comment]
@micler Oh, and don't use `fullresolution` to force a specific exclusive fullscreen mode. That path is kinda broken, you should use `desktop` or not set it at all.

Remove that line from your config, or use `fullresolution = desktop`, remove the vsync line as well, then report back please.

**Comment by Grounded0** [comment]
Connected issues and PRs:

- #3206

**Comment by micler** [comment]
Thanks to all of you for the tips, i'll try to answer to all:

First about my setup: I'm using a desktop, I have a Ryzen 3600X CPU + RX5700 GPU, using a 32" 1440p monitor (Freesync).
I don't have anything specific set for dosbox, the only active option I have set is "Radeon chill", at 62fps, and it's a global option.

I didn't reuse an old config (was using dosbox ECE before, but many options were obsolete), so I took the exemple from the repo and customized a few lines, like the resolution: on dosbox ECE I had "original" as a resolution, which then translated to 640x400, so I set 640x400 in Staging (and in viewport). I can see in the logs that it transforms to 640x480. (I changed the value, but setting 640x480 doesn't have any impact on my issue.)
I also changed the cycles to 20000, which is what I used in dosbox ECE.

I tried fullresolution=desktop, but then the game doesn't fill the screen. And the initial problem remains.

I don't run many old games, i'm using dosbox for F1GP and GP2, and the issue is the same. (I use the same config except cycles)

Following your remarks I did a lot of testing: using the options @interloper98 mentionned, the problem disappeared. So I tried option by option (but keeping vsync off), and found that the magic one is "dos_rate". If I leave it to auto, frames are dropped, if I set it to 60, all is ok.
Following this, I wondered if the other parameter mentionned,  presentation_mode = cfr, would do the trick, but it does not.
I'm wondering if there could be an incompatibility between Freesync/VRR and the frame rate variation in dosbox ? When in auto the Frame Rate seen by the monitor oscillates between 70 and 72. In windowed mode, the monitor stays locked at 60Hz (Windows's value).

I attached a log with my old "broken" configuration.
[dosbox_log.txt](https://github.com/user-attachments/files/16216198/dosbox_log.txt)

**Comment by johnnovak** [comment]
@micler glad to hear you're getting results you like with the suggested settings.

However, VRR should definitely work for you on your FreeSync monitor. Forcing `dos_rate = 60` maybe "fixes" the problem for you (more like masks it), but that's a hack as 320x200 VGA DOS games run at 70 Hz. `dos_rate = 60` breaks many games as it forces them to run at 60 Hz, but as most assume standard 70 Hz VGA, that can break their internal timings and strange things or crashes might happen. Therefore, forcing `dos_rate` is  not recommended for general use. It might work fine with this game, but you'll definitely run into problems sooner or later with other titles.

But the good news is we can most likely do better on your nice FreeSync VRR monitor! If you'd like to improve the situation, here's a few things to try.

Let's address some of your comments first.

> I have set is "Radeon chill", at 62fps, and it's a global option.

Ok, so AFAIK that acts as a dynamic frame limiter. Setting this to 62 FPS is very bad news for DOSBox as the native VGA rate is a bit above 70 Hz for all 320x200 DOS games! If this works how I think it works, it will probably interfere with DOSBox in an undesirable way and it might be the source of your problems. Please temporarily disable Radon Chill altogether. Then if you're getting the results you want, you can try setting it again, but a higher rate, say 75 or 80 Hz or higher.

> I tried fullresolution=desktop, but then the game doesn't fill the screen (need to adapt the viewport value i guess). And the initial problem remains.

One more time, please leave `fullresolution` alone, at least until we figure out the best settings for your VRR setup. With the default `fullresolution=desktop` you're running Staging at the native resolution of your monitor—that's what you want, especially if you want to take advantage of the CRT shaders. Those work best at monitor's native res.

Again, setting `fullresolution` to _any_ other value than `desktop` is _not_ supported; it's a legacy setting that is broken, we don't test it, and it's very low priority for us to fix it. Running at the monitor's native res is what we recommend.

If you want the image to fill the screen, `integer_scaling = off` is the solution, which should work well on your 1440p monitor up to 640x480 DOS res.  If you don't want CRT shaders, set `glshader = sharp` too.

Also, `output = texture` is a fallback option, so leave `output` alone too and use the default `opengl` setting (which is required for the CRT shaders anyway).

> When in auto the Frame Rate seen by the monitor oscillates between 70 and 72. In windowed mode, the monitor stays locked at 60Hz (Windows's value).

Yes, that's all normal. In windowed mode the Window's desktop's refresh rate is in effect, which is 60 Hz. But the proper VGA rate is a bit above 70 Hz, so your monitor is telling you the truth in fullscreen 😄 

---

So, back to your problem, according to the logs, your VRR monitor is not detected properly; Staging sees it as a fixed 60 Hz display. I'd bet good money on this this is because the Radon Chill stuff.

```
SDL: Using standard SDI display refresh rate of 60 Hz
```

So please disable Radeon Chill altogether, and try the below minimal config. After this, the logs should say a VRR display has been detected.

If still no cigar, please set `host_rate = vrr` manually.  Then if `host_rate = vrr` is not enough, try adding `presentation_mode = cfr` as well.

So:

1. Disable Radon Chill
2. Use the following minimal config (don't add `dos_rate` or anything else).
3. Add in `host_rate` and `presentation_mode` if needed.

```ini
[sdl]
# host_rate = vrr
# presentation_mode = cfr

[render]
# to make the image fill the screen
integer_scaling = off

# set this if you don't like CRT shaders
# glshader = sharp

[cpu]
cycles = 20000
cycleup = 2000
cycledown = 2000

[autoexec]
# command to start the game
```

I'm really curious if this solves your problem. If it does, this is the ideal way to utilise your VRR monitor with Staging.

I don't have a VRR monitor, so your feedback would be valuable, especially the info whether we should recommend people to disable frame limiters such as Radeon chill to get VRR working properly.

**Comment by micler** [comment]
Ok, I pretty much tried everything, here are the results:

Setting everything in auto with RadeonChill off works. (FPS rate oscillates between 70 and 90, goes up to 144 on still images)
Using VRR as host_rate works.
Using CFR as presentation mode is glitchy. (The FPS seen by the screen oscillate between 80 and 144 al the time)
Using sdi doesn't work, display is jerky
Activating RadeonChill doesn't work, it's jerky.

So as you guessed, it does work with default settings. Thanks for the explanation of the various parameters/behaviors btw, it's very interesting.

The only drawback is that VRR isn't always activated. As I mentionned earlier, my default rate in windows is 60Hz. Sometimes, when switching Staging to fullscreen the rate switches to VRR (and doesn't switch back when exiting the app or getting back to windowed mode btw). But sometimes it doesn't. Which made me feel like I was crazy because I didn't understand why the same config didn't work everything, until I checked the display rate in Windows options every time.
I can cope with that bug, if switch the display rate to 144 in windows before starting Staging, it works.

**Comment by MasterO2** [comment]
> Ok, I pretty much tried everything, here are the results:
> 
> Setting everything in auto with RadeonChill off works. (FPS rate oscillates between 70 and 90, goes up to 144 on still images) Using VRR as host_rate works. Using CFR as presentation mode is glitchy. (The FPS seen by the screen oscillate between 80 and 144 al the time) Using sdi doesn't work, display is jerky Activating RadeonChill doesn't work, it's jerky.
> 
> So as you guessed, it does work with default settings. Thanks for the explanation of the various parameters/behaviors btw, it's very interesting.
> 
> The only drawback is that VRR isn't always activated. As I mentionned earlier, my default rate in windows is 60Hz. Sometimes, when switching Staging to fullscreen the rate switches to VRR (and doesn't switch back when exiting the app or getting back to windowed mode btw). But sometimes it doesn't. Which made me feel like I was crazy because I didn't understand why the same config didn't work everything, until I checked the display rate in Windows options every time. I can cope with that bug, if switch the display rate to 144 in windows before starting Staging, it works.

Can you show us your new configuration file now?

**Comment by micler** [comment]
Here it is:
[dosbox-staging.txt](https://github.com/user-attachments/files/16239958/dosbox-staging.txt)

**Comment by Grounded0** [comment]
Most likely not a Staging bug because VRR has been tested by some real freaks so we got it right and has been in use for 2 years.

Have you applied newest firmware update for your display?

**Comment by interloper98** [comment]
I was reading this wiki page - https://github.com/dosbox-staging/dosbox-staging/wiki/High-refresh-rate-gaming

Then looked up [the specs](https://www.lg.com/us/monitors/lg-32GK850F-B-gaming-monitor) about your display.  Comments at the bottom mention: _"The LG 32" Class QHD Gaming Monitor with Radeon FreeSync™ 2 Technology (31.5" Diagonal), Model # 32GK850F-B **has a Freesync range of 72 - 144 Hz via DisplayPort as shown on the AMD site**"_

And here [on reddit](https://www.reddit.com/r/Monitors/comments/935iba/lg_32gk850f_page_is_up_1440p_144_hz_32_va_hdr400/): _"In order to get LFC, you still need to trigger the non-standard range, which means potential flickering and other issues. YMMV"_

The game sets and uses this mode: `VGA 320x200 256-colour graphics mode 13h at 70.086 Hz`, so it should be impossible for DOSBox Staging to produce frames faster than 70.086 fps. 

You reported seeing rates even higher - 80+ fps all the way to 144 fps, so perhaps what those redditors mentioned about triggering the LFC (low frame-rate compensator) is happening here: F1GP goes from the 20s to the 40s and higher (perhaps maxing out at 70 fps?), so when it's "low" the monitor is using some multiplier (say: 25 fps by 3x = 75 fps) but then the game briefly jumps to 50 fps where that 3x multiplier is suddenly too much! So it steps back to a 2x multiplier.

The redditors mention you can get flicker and some problems with the LFC constantly adjusting.

I know you mentioned finding some workable settings (and that default is good) - so this is more just me sharing some thoughts -- I would try to get the game (and DOSBox Staging) itself producing a fixed frame-rate to land inside 72 to 144 range 100% of the time. 
 - `presentation_mode = cfr` combined with `dos_rate = 80` (or something like that), and then give the game lots of cycles Perhaps 120000.
 - Another line of thinking is to work with the LFC by locking the game to a nice but low rate so the LFC can use a fixed multiple and not hunt around.
 - `presentation_mode = cfr` combined with `dos_rate = 40`, also with enough cycles. Maybe the display would do 3x multiple and will run at a steady 120 Hz.

**Comment by johnnovak** [comment]
@interloper98  Well, that's a good reminder to _never ever_ buy a VRR monitor that cannot sync down to below 50 Hz rates 😄 But yeah, the multiplies probably work, so 100 Hz instead of the native 50 Hz, 120 Hz instead of the native 60 Hz, etc.

Again, about those `dos_rate` recommendations, please note that `dos_rate` _is a hack_! Works with some games, horribly breaks other games as it _can_ mess up the game's timing and the audio. Generally, you should leave `dos_rate` alone, and only mess with it in a select few games (e.g., some late 90s FPS games) if you want to achieve something specific. It's definitely power-user territory. People should _never_ touch `dos_rate` just fix host OS / monitor refresh rate issues!

It's more fruitful to experiment with `host_rate` instead (if you want to experiment with anything, not just accept the default auto-determined settings). Please read the config descriptions for `dos_rate` and `host_rate`. So you can replace `dos_rate` with `host_rate` in those config ideas @micler. But then, you'd need to reconfigure `host_rate` per game, potentially, which is a pain.  Best to leave it on auto.

I'm also interested in figuring out the root cause for the "flakiness" of the VRR detection. Do you always get proper VRR output if you set `host_rate = vrr` instead of leaving it on `auto`?

I will introduce a new feature in a couple of days to log extra info about the "auto" rate/vsync detection, perhaps you could try that too. I will ping you about it if you're interested.

**Comment by interloper98** [comment]
> that's a good reminder to never ever buy a VRR monitor that cannot sync down to below 50 Hz rates

Lol that's for sure!
(This model @micler has is now > 5 years old, I think pretty close to first gen freesync.. or just after)

> Again, about those dos_rate recommendations, please note that dos_rate is a hack! 

I should stop using it 😆 ; at least some games are quite compliant when  their rate is shoe horned either up or down it to fit these picky displays! (Two wrongs might make a right - a bit more playable game on first gen freesync 🤔 😄 )

> I'm also interested in figuring out the root cause for the "flakiness" of the VRR detection.

Yeah, seeing  how the code checks the display's advertised refresh rate again a threshold; maybe when @micler set the AMD driver to run at 60 Hz, I guess DOSBox Staging would see the display as a standard (non-VRR). Maybe now that @micler has bumped it up to 144 Hz, it (hopefully?) should always be detected as VRR. But answering your questions should get to the bottom of it!

**Comment by johnnovak** [comment]
> Yeah, seeing how the code checks the display's advertised refresh rate again a threshold; maybe when @micler set the AMD driver to run at 60 Hz, I guess DOSBox Staging would see the display as a standard (non-VRR).

I'm 99% sure that's the case, yeah. Frame limiters are a bad idea for home computer and console emulators that emulate a specific fixed refresh rate. Modern games work very differently than emulators, and frame limiters were invented for them.

> Maybe now that @micler has bumped it up to 144 Hz, it (hopefully?) should always be detected as VRR. But answering your questions should get to the bottom of it!

Yeah, I wanna find out whether `host_rate = auto` is flaky, so it doesn't always pick `vrr`, or `host_rate = vrr` is also unreliable (which is unlikely, but better to test and see).

**Comment by Grounded0** [comment]
Install LG OnScreen Control to update firmware of your display:

https://www.lg.com/uk/support/product-support/cs-32GK850F-B.AEK/

LG Monitor - How to Update On Screen Control Firmware:

https://www.lg.com/us/support/help-library/lg-monitor-how-to-update-on-screen-control-firmware--20153265521846

**Comment by micler** [comment]
I had the same ideas to check if the VRR glitch could be fixed yesterday but didn't had time.
I really was full of hope, but alas...
Host_rate = vrr : doesn't change a thing.
I even tried adding presentation mode = vfr, still 60Hz.

I tried starting in full screen, still  locked at 60Hz. I think I've been mixing things up yesterday, thinking it worked some times. Today, it does not if the screen is set at 60Hz. And works everytime with the screen set to 144. You might be right with the VRR detection.

I'm currently downloading the update for the On screen display, thanks for the link. It's downloading @30ko/s... download already failed once. I'll have to be patient, it feels like 1999. I'll send an update if the download actually finishes... 

The screen works from 48Hz to 144 quite smoothly, I'm using the Extented Freesync option. (I tried once settings the Radeon chill to 50fps, the screen did work at such a "low" frame rate)

Regarding F1GP, it can work from 8.1 to 25 fps (that was considered smooth at the time ^^) with a vanilla install, a hack allows higher rates, I've set my install to 30 fps. (the physics engine can create glitches above 40fps) So it doesn't run even close from dosbox refresh rate.

**Comment by micler** [comment]
Finally fetched the LG application and updated the screen.  I now have a nice FreeSync2 logo in the display.
Some behavior regarding Staging.

**Comment by johnnovak** [comment]
> edit: In fact, I see a change in behavior, and I don't get it: the monitor refresh rates stays at 60Hz, but i don't have any jerkiness in the game anymore, that's unbelievable !
> I'm lost regarding the explanation.

With all vsync, host_rate, dos_rate, and presentation_mode set to auto? That's good outcome.

Frankly, I would not put too much faith in the FPS and Hz figures reported by the monitor either in the hardware OSD on in the Windows-side drivers when it comes to VRR. If it only reports a single FPS or Hz value, it must do some polling, sampling, and averaging, and no matter how it does it, it will always only be an approximation of the truth. A continuous FPS timeline would be the real deal.

But even then things get murky, because there's our presentation intervals (our presentation FPS), and what the drivers and monitor do with it.

So, if it works, it works 😄 But now the logs say VRR is auto-selected, right?

**Comment by Grounded0** [comment]
I've spent 4 years in the trenches of this issue tracker so you're bound to become good in sorting these out.

**Comment by interloper98** [comment]
![8xacv8](https://github.com/user-attachments/assets/20c29fc3-30cf-40ab-943c-5c6785b2d99f)

> I've spent four years in the trenches of this issue tracker

@Grounded0 , that's some battle knowledge! I had no idea monitors even had firmware that could be upgraded!

**Comment by johnnovak** [comment]
> @Grounded0 , that's some battle knowledge! I had no idea monitors even had firmware that could be upgraded!

Lol, me neither 😆  Guess what I'll do today when I get home from work... (hopefully won't have to search for "help fix bricked monitor firmware" on my wife's laptop afterwards 😅)

**Comment by micler** [comment]
> > edit: In fact, I see a change in behavior, and I don't get it: the monitor refresh rates stays at 60Hz, but i don't have any jerkiness in the game anymore, that's unbelievable !
> > I'm lost regarding the explanation.
> 
> With all vsync, host_rate, dos_rate, and presentation_mode set to auto? That's good outcome.
> 
> Frankly, I would not put too much faith in the FPS and Hz figures reported by the monitor either in the hardware OSD on in the Windows-side drivers when it comes to VRR. If it only reports a single FPS or Hz value, it must do some polling, sampling, and averaging, and no matter how it does it, it will always only be an approximation of the truth. A continuous FPS timeline would be the real deal.
> 
> But even then things get murky, because there's our presentation intervals (our presentation FPS), and what the drivers and monitor do with it.
> 
> So, if it works, it works 😄 But now the logs say VRR is auto-selected, right?

Yes indeed: as the log shows:
SDL: Using VRR-adjusted display refresh rate of 57 Hz
DISPLAY: VGA 720x400 16-colour text mode 03h at 70.087 Hz VFR, scaled to 640x356 pixels with 1:1 (1:1) pixel aspect ratio

**Comment by micler** [comment]
> ![8xacv8](https://private-user-images.githubusercontent.com/165201780/349344777-20c29fc3-30cf-40ab-943c-5c6785b2d99f.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEzMzIwMTksIm5iZiI6MTcyMTMzMTcxOSwicGF0aCI6Ii8xNjUyMDE3ODAvMzQ5MzQ0Nzc3LTIwYzI5ZmMzLTMwY2YtNDBhYi05NDNjLTVjNjc4NWIyZDk5Zi5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxOFQxOTQxNTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zZjdhNjY0ZWZiYTQ3NzhkODBlMWVmMDlmY2IyZWZhZDA4MWU3Mjc1YjYzODMwOTU1ZGFmZDhmNWM2MTQwOTEyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.JmlK95jM5yla9VVdQeMs57VbU2LDJRtgWgt5eIUR2Go)
> 
> > I've spent four years in the trenches of this issue tracker
> 
> @Grounded0 , that's some battle knowledge! I had no idea monitors even had firmware that could be upgraded!

Ok, there's one weird thing though. The game is fine, VRR seems to work, by the frame rate never goes higher than 60fps. (Windows' limit)
So I put 144Hz in Windows to see the difference, and Oh my god that's terrible. The display is jerky as hell, and the frame rate oscillates likes crazy between 48 and 110hz.

On one side I'm happy to have found a way to make things work... but I definitely don't understand some behaviors.
When using 144Hz on windows, the log states:

> SDL: Using VRR-adjusted display refresh rate of 141 Hz

**Comment by micler** [comment]
Serioulsy, this issue is beginning to feel like a big joke.

So I restarted my computer and made sure my screen was at 144Hz. I was willing to do some more tests to trying understand why it was jerky @144hz. I launched the game again, in fullscreen, and all was fine. But the frame rate at 60Hz, and my parameter in Windows had changed !
I check several time. As soon as Staging goes fullscreen, the rate switches to 60hz.

Now I must mention I have a second display, roughly the same as the first one. (GK850 vs GK650). My second display had not been updated. I set Staging on the second screen. The frame rate was not locked to 60hz (betwen 75 and 90), and it was smooth. As was the first screen before the update.

I'm completely lost. Why did the firmware update + a reboot gave the right the Staging to change my Windows settings ?
So, one more test: I locked my second screen to 60Hz, and launched Staging on it.
In the log this time, the SDL library initialises with SDI at 60Hz. And jerkiness returns. At least, this is consistent with the first results I got.

**Comment by johnnovak** [comment]
Well, I'm inclined to say this is a Windows / monitor firmware / GPU driver issue, and we can't possibly test every setup out there. So maybe just leave this at this this:

- VRR works for most people,
- auto mode tries to use VRR and succeeds most of the time (unless some weird frame limiters are in use),
- if auto mode fails, the user has enough levers to find a workable setup by setting the video config options manually.

It looks like Staging is not at fault here; your problems seem to be OS and hardware related. Difficult to tell without attaching a "known good VRR monitor make" to your computer and re-test with that.

I'm not trying to be dismissive, just we're starting to exhaust our remote troubleshooting powers without having access to your setup and us core members owning no VRR displays.

If we were a professional operation, we'd have a dozen or two laptops from the last 10-15 years with various GPUs, and a dozen VRR monitors of various makes to test things thoroughly on all OSes with different GPUs... but we are only hobbyists.

**Comment by johnnovak** [comment]
@micler Also, setting 144 Hz in Windows sounds fishy. I don't have a VRR monitor, but that just sounded wrong, so I googled it and it turns out you can enable VRR in Win 10/11 without setting fixed rates. So maybe that the root cause of all your problems, forcing your display to use a fixed rate?

https://www.elevenforum.com/t/enable-or-disable-variable-refresh-rate-for-games-in-windows-11.12052/

**Comment by Grounded0** [comment]
Suggest closing this. We're only interested are we screwing up something within the code we support. This project is mainly funded off from one guys disability pension so we don't have the resources to explore this further.

**Comment by micler** [comment]
@johnnovak: well, you have to choose a refresh rate for Windows, most of the game will set their own when they start, when they do support VRR. (I have not witness any issue on the other games I'm using, both old and recent, I'm been also using PS1 n PS2 emulators without encountering issues)
I agree with you, i don't think staging is at fault here, i'm a bit puzzled by the change of behavior when updating the firmware, but as I said before, I found a working setup so I'm happy. I definitely don't expect anything more than the support provided.

I'd like to thank you all for the time you spent on my issue, and thank you again for the advice and insight you could give me on Staging.

Any of you actually contacted GOG for partnership / contributions ?

**Comment by micler** [comment]
Thanks !

**Comment by johnnovak** [comment]
> I agree with you, i don't think staging is at fault here, i'm a bit puzzled by the change of behavior when updating the firmware, but as I said before, I found a working setup so I'm happy. I definitely don't expect anything more than the support provided.

Great to hear that 🥳 

> I'd like to thank you all for the time you spent on my issue, and thank you again for the advice and insight you could give me on Staging.

No problem, we're happy to help. We hope you'll enjoy Staging 😎 I've also learned something valuable from this discussion regarding frame limiters. 👍🏻 

> Any of you actually contacted GOG for partnership / contributions ?

The number of times we've been contacted on such matters is exactly zero 😆
